src/storage/backend/backend_impl.rs

```diff
@@ -59,17 +68,42 @@ impl BackendBuilder {
                 ref access_key_id,
                 ref secret_access_key,
                 ref bucket_name,
+                ref region,
+                ref max_concurrent_requests,
             }) => {
                 let mut builder = S3::default();
 
                 builder
                     .endpoint(endpoint_url)
                     .access_key_id(access_key_id)
                     .secret_access_key(secret_access_key)
-                    .region("auto")
                     .bucket(bucket_name);
 
-                Operator::new(builder)?.layer(layer).finish()
+                // Init region
+                if region.is_some() {
+                    builder.region(region.as_ref().map_or("auto", String::as_str));
+                } else {
+                    // Auto detect region
+                    let region = match S3::detect_region(endpoint_url, bucket_name).await {
+                        Some(region) => region,
+                        None => "auto".to_owned(),
+                    };
+                    builder.region(region.as_str());
+                }
```

Comment:
Please use `if let` or `match` here.

Reply:
ok :)

Reply:
Oh, I mean, you can code something like:
```rust
if let Some(region) = region {
    builder.region(region);
} else {
    let region = S3::detect_region(endpoint_url, bucket_name).await.unwrap_or("auto");
    builder.region(region);
}
```

Reply:
fixed.